Webapp/sources/blueprints/manage_workgroup/routes.py

```diff
@@ -344,3 +347,117 @@ def go_to_workgroup(workgroup: str) -> Response:
             "manage_workgroup.manage_workgroup", has_request="yes", workgroup=workgroup
         )
     )
+
+
+@manage_workgroup_bp.route("/add_user_by_email", methods=["GET", "POST"])
+@principal_investigator_required
+def add_user_by_email(workgroup):
+    email = request.args.get("email")
+    user_type = request.args.get("user_type")
+
+    user = services.user.from_email(email)
+
+    new_role_display_string = workgroup_dict[user_type]["display_string"]
+
+    added_person = services.person.from_email(email)
+    if not added_person:
+        flash(f"User with email: {email} does not exist! Please try again.")
+
+    wg = services.workgroup.from_name(workgroup)
+    if not services.workgroup.get_user_type(workgroup, user):
+
+        notification = models.Notification(
+            person=added_person.id,
+            type="You Have Been Added to a Workgroup",
+            info="A Principal Investigator has added you to the Workgroup, "
+                 + workgroup
+                 + ", as a "
+                 + new_role_display_string
+                 + ". Please respond below.",
+            time=datetime.now(),
+            status="active",
+            wg=wg.name,
+            wb="",
+            wg_request="added",
+        )
+
+        duplicates = (
+            db.session.query(models.WGStatusRequest)
+            .join(models.Person, models.WGStatusRequest.person == models.Person.id)
+            .join(models.User)
+            .filter(models.User.email == user.email)
+            .join(models.WorkGroup)
+            .filter(models.WorkGroup.id == wg.id)
+            .filter(models.WGStatusRequest.status == 'active')
+            .all()
+        )
+
+        if duplicates:
+            flash(
+                "This User's membership has already been requested for this Workgroup."
+            )
+
+        else:
+
+            services.email.send_notification(added_person)
+            db.session.add(notification)
+            db.session.commit()
+
+            wg_request = models.WGStatusRequest(
```

Comment:
Could some of these queries go in a services file? A few look similar

Reply:
I've moved these over in a future PR, accidentally committed the changes to the wrong branch but it will be done soon